includes/models/model.llms.student.php

```diff
@@ -1841,4 +1909,162 @@ private function update_completion_status( $status, $object_id, $object_type, $t
+		 */
+		return apply_filters( 'llms_is_' . $object_type . '_favorite', $ret, $object_id, $object_type, $this );
+
+	}
+
+	/**
+	 * Mark a lesson favorite for the given user.
+	 *
+	 * @since [version]
+	 *
+	 * @see llms_mark_favorite() calls this function without having to instantiate the LLMS_Student class first.
+	 *
+	 * @param int    $object_id   WP Post ID of the Lesson.
+	 * @param string $object_type Type, 'Lesson'.
+	 * @return bool
+	 */
+	public function mark_favorite( $object_id, $object_type ) {
+
+		// Short circuit if it's already favorited.
+		if ( $this->is_favorite( $object_id, $object_type ) ) {
+			return true;
+		}
+
+		return $this->update_favorite_status( 'favorite', $object_id, $object_type );
+
+	}
+
+	/**
+	 * Mark a lesson unfavorite for the given user.
+	 *
+	 * @since [version]
+	 *
+	 * @see llms_mark_unfavorite() calls this function without having to instantiate the LLMS_Student class first.
+	 *
+	 * @param int    $object_id   WP Post ID of the Lesson.
+	 * @param string $object_type Type, 'Lesson'.
+	 * @return bool
+	 */
+	public function mark_unfavorite( $object_id, $object_type ) {
+
+		// Short circuit if it's not favorited.
+		if ( ! $this->is_favorite( $object_id, $object_type ) ) {
+			return true;
+		}
+
+		return $this->update_favorite_status( 'unfavorite', $object_id, $object_type );
+
+	}
+
+	/**
+	 * Triggers actions for favorite/unfavorite.
+	 *
+	 * Update the favorite status of a lesson for the current student.
+	 * Inserts / updates necessary user postmeta data.
+	 *
+	 * @since [version] Use filterable functions to determine if the object can be marked favorite.
+	 *
+	 * @param string $status      New status to update to, either "favorite" or "unfavorite".
+	 * @param int    $object_id   WP Post ID of the Lesson.
+	 * @param string $object_type Type, 'Lesson'.
+	 * @param string $trigger     String describing the reason for the status change.
+	 * @return bool
+	 */
+	private function update_favorite_status( $status, $object_id, $object_type, $trigger = 'unspecified' ) {
+
+		$student_id = $this->get_id();
+
+		/**
+		 * Fires before a student's object favorite status is updated.
+		 *
+		 * The dynamic portion of this hook, `$status`, refers to the new completion status of the object,
+		 * either "favorite" or "unfavorite".
+		 *
+		 * @since [version]
+		 *
+		 * @param int    $student_id  WP_User ID of the student.
+		 * @param int    $object_id   WP Post ID of the Lesson.
+		 * @param string $object_type Type, 'Lesson'.
+		 * @param string $trigger     String describing the reason for the status change.
+		 */
+		do_action( "before_llms_mark_{$status}", $student_id, $object_id, $object_type, $trigger );
+
+		// Insert / Remove meta data.
+		if ( 'favorite' === $status ) {
+			$this->insert_favorite_postmeta( $object_id, $trigger );
+		} elseif ( 'unfavorite' === $status ) {
+			$this->remove_unfavorite_postmeta( $object_id, $trigger );
```

Comment:
I missed this :D @actuallyakash 
what's remove unfavorite eheh, I think you meant to go with
`remove_favorite_postmeta()`

Reply:
Sometimes My Genius Is Almost Frightening 😶‍🌫️
fixed.

Reply:
Sometimes my genius is almost frightening 😶‍🌫️
fixed.